File: test/plugin/spring/fixtures/general/src/main/java/inga/server/RequestMappingDefinition.java
package p1;

import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RestController;

@RequestMapping("/mapping")
@RestController
public class RequestMappingDefinition {
    @RequestMapping
    public void allWithNoValue() {
    }

    @RequestMapping(value = "/{v}", method = RequestMethod.GET)
    public void getWithValue(@PathVariable String v) {
    }

    @RequestMapping(value = {"/{v1}", "/{v1}/{v2}"}, method = RequestMethod.GET)
    public void getWithValues(@PathVariable String v1, @PathVariable(required = false) String v2) {
    }

    @RequestMapping(path = "/{v}", method = RequestMethod.GET)
    public void getWithPath(@PathVariable String v) {
    }

    @RequestMapping(path = {"/{v1}", "/{v1}/{v2}"}, method = RequestMethod.GET)
    public void getWithPaths(@PathVariable String v1, @PathVariable(required = false) String v2) {
    }
}

File: test/plugin/spring/fixtures/general/src/main/kotlin/inga/server/RequestMappingDefinition.kt
package inga.server

import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestMethod
import org.springframework.web.bind.annotation.RestController

@RequestMapping("/mapping")
@RestController
class RequestMappingDefinition {
    @RequestMapping
    fun allWithNoValue() {
    }

    @RequestMapping(value = ["/{v}"], method = [RequestMethod.GET])
    fun getWithValue(@PathVariable v: String) {
    }

    @RequestMapping(value = ["/{v1}", "/{v1}/{v2}"], method = [RequestMethod.GET])
    fun getWithValues(@PathVariable v1: String, @PathVariable(required = false) v2: String) {
    }

    @RequestMapping(path = ["/{v}"], method = [RequestMethod.GET])
    fun getWithPath(@PathVariable v: String) {
    }

    @RequestMapping(path = ["/{v1}", "/{v1}/{v2}"], method = [RequestMethod.GET])
    fun getWithPaths(@PathVariable v1: String, @PathVariable(required = false) v2: String) {
    }
}
